homeassistant/components/sun/__init__.py

```diff
@@ -120,15 +120,20 @@ def point_in_time_listener(self, now):
         """Run when the state of the sun has changed."""
         self.update_sun_position(now)
         self.update_as_of(now)
-        self.async_schedule_update_ha_state()
+        self.async_write_ha_state()
+        state = self.hass.states.get('sun.sun')
```

Comment:
Don't do this. Entities should not add a dependency on their own state in the state machine. All the info should be there locally.

Reply:
yeah, this was really lazy..
Will fix!